File: app/src/main/java/com/elyeproj/myapplication/MainActivity.java
package com.elyeproj.myapplication;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements MainView {

    private MainPresenter mainPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainPresenter = new MainPresenter(this);
        mainPresenter.fetchData();
    }

    @Override
    public void showResult(String data) {
        Toast.makeText(this, data, Toast.LENGTH_LONG).show();
    }
}

File: app/src/main/java/com/elyeproj/myapplication/MainActivity.kt
package com.elyeproj.myapplication

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

class MainActivity : AppCompatActivity(), MainView {

    private var mainPresenter: MainPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mainPresenter = MainPresenter(this)
        mainPresenter!!.fetchData()
    }

    override fun showResult(data: String) {
        Toast.makeText(this, data, Toast.LENGTH_LONG).show()
    }
}
